support-frontend/assets/pages/weekly-subscription-checkout/helpers/deliveryDays.ts

```diff
@@ -59,14 +65,58 @@ function getTierThreeDeliveryDate(today?: number) {
 	return result;
 }
 
-function getGuardianAdLiteDate(today?: number) {
-	const firstValidDate = addDays(new Date(today ?? new Date()), 15);
-	return firstValidDate;
-}
+const productDeliveryOrStartDate = (
+	productKey: ActiveProductKey,
+	paperProductOptions?: ActivePaperProductOptions,
+): Date | undefined => {
+	switch (productKey) {
+		case 'GuardianAdLite':
+			return addDays(new Date(), 15);
+		case 'TierThree':
+			return getTierThreeDeliveryDate();
+		case 'NationalDelivery':
+		case 'HomeDelivery':
+		case 'SubscriptionCard': {
+			// paper productOption undefined check
+			if (paperProductOptions === undefined) {
+				return undefined;
+			}
+			const paperDeliveryDate =
+				productKey === 'SubscriptionCard'
+					? getPaymentStartDate(Date.now(), paperProductOptions)
+					: productKey === 'HomeDelivery'
+					? getHomeDeliveryDays(Date.now(), paperProductOptions)[0]
+					: getVoucherDays(Date.now(), paperProductOptions)[0];
+			// paper home and voucher delivery date array empty check
+			if (paperDeliveryDate === undefined) {
+				return undefined;
+			}
+			return paperDeliveryDate;
```

Comment:
I think this could be simplified to just `return paperDeliveryDate` since if it's undefined, we return undefined :-)